server/tests/participant.form.test.js

```diff
@@ -271,19 +266,42 @@ describe('Participants Service', () => {
     );
   });
 
+  it('Status change happy path', async () => {
+    const employerAId = v4();
+    const participants = await getParticipants({ isEmployer: true, id: employerAId, regions }, null, null, null, null, ['open']);
+    const participantId = participants.data[0].id;
+
+    // Engage, reject
+    expect((await setParticipantStatus(employerAId, participantId, 'prospecting')).status).toEqual('prospecting');
+    expect((await setParticipantStatus(employerAId, participantId, 'rejected', { final_status: 'not responsive', previous: 'prospecting' })).status).toEqual('rejected');
+
+    // Engage, withdraw
+    expect((await setParticipantStatus(employerAId, participantId, 'prospecting')).status).toEqual('prospecting');
+    expect((await setParticipantStatus(employerAId, participantId, 'rejected', { final_status: 'withdrawn', previous: 'prospecting' })).status).toEqual('rejected');
+
+    // Re-engage, interview, reject
+    expect((await setParticipantStatus(employerAId, participantId, 'prospecting')).status).toEqual('prospecting');
+    expect((await setParticipantStatus(employerAId, participantId, 'interviewing')).status).toEqual('interviewing');
+    expect((await setParticipantStatus(employerAId, participantId, 'rejected', { final_status: 'not qualified', previous: 'interviewing' })).status).toEqual('rejected');
+
+    // Re-engage. offer made, reject
+    expect((await setParticipantStatus(employerAId, participantId, 'prospecting')).status).toEqual('prospecting');
+    expect((await setParticipantStatus(employerAId, participantId, 'interviewing')).status).toEqual('interviewing');
+    expect((await setParticipantStatus(employerAId, participantId, 'offer_made')).status).toEqual('offer_made');
+    expect((await setParticipantStatus(employerAId, participantId, 'rejected', { final_status: 'position filled', previous: 'offer_made' })).status).toEqual('rejected');
+
+    // Re-engage, hire
+    expect((await setParticipantStatus(employerAId, participantId, 'prospecting')).status).toEqual('prospecting');
+    expect((await setParticipantStatus(employerAId, participantId, 'interviewing')).status).toEqual('interviewing');
+    expect((await setParticipantStatus(employerAId, participantId, 'offer_made')).status).toEqual('offer_made');
+    expect((await setParticipantStatus(employerAId, participantId, 'hired')).status).toEqual('hired');
+  });
+
   it('Employer A hires participant X then employer B cannot hire participant X', async () => {
-    const employerAId = '12345-a';
-    const employerBId = '12345-b';
-
-    const regions = [
-      'Fraser',
-      'Interior',
-      'Northern',
-      'Vancouver Coastal',
-      'Vancouver Island',
-    ];
+    const employerAId = v4();
+    const employerBId = v4();
 
-    const participants = await getParticipants({ isEmployer: true, id: employerAId, regions });
+    const participants = await getParticipants({ isEmployer: true, id: employerAId, regions }, null, null, null, null, ['open']);
```

Comment:
I added this filter to almost all of the tests to prevent them from getting participants who have actually been marked as hired in previous test executions